### ISSUE TITLE
add typehints

### ISSUE THREAD

**Issue by jf---** [LaughlinResearch/AFEM#11]
as it says on the tin -- I think typehints would be a welcome addition. 
@trelau would you accept such a PR, since that would basically make the project python>=3.6+?

**Comment by trelau** [comment]
Yes, I think so. AFEM depends on pyOCCT which is only supported for 3.6 and above.

As a side note, I've been trying to migrate pyOCCT and pySMESH to latest versions (OCCT 7.5.0 and SMESH 9.6.0) but hit a snag here https://github.com/trelau/SMESH/issues/37. Note I also broke out pySMESH from pyOCCT.

**Comment by jf---** [comment]
Great to see the 7.5 update is nearby... The smesh breakout is a solid idea...
Great, I'll horse around wrt type hints... 

I understand that the issue wrt SMESH is mostly osx related?

**Comment by trelau** [comment]
Yes it seems osx related and so far has me stumped (and i don't have a local osx system to quickly debug...)

**Comment by jf---** [comment]
Understood. Is this keeping you from releasing the breakout version of (pay)smesh & pyocct (occt 7.5)?

**Comment by trelau** [comment]
Yes. I’d like to get pyocct, smesh, and pysmesh fully updated to latest versions of OCCT (7.5) and Salome (9.6) and update the conda feedstock for SMESH. Then refresh afem with new libraries.

**Comment by SkyFishers** [comment]
Hello,it cant work in windows now ,I install the library by conda,it cant find mesh.dll.
